File: BE-Task_1-Ruby/internal/domain/entities/purchase.rb
module RefundBackend
  module Domain
    module Entities
      class Purchase
        attr_reader :category, :purchase_id, :purchase_timestamp, :region

        def initialize(purchase_id:, region:, category:, purchase_timestamp:)
          @purchase_id = purchase_id
          @region = region
          @category = category
          @purchase_timestamp = purchase_timestamp
          freeze
        end

        def to_h
          {
            purchase_id: purchase_id,
            region: region,
            category: category,
            purchase_timestamp: purchase_timestamp
          }
        end
      end
    end
  end
end

File: BE-Task_1-Php/internal/domain/entities/purchase.php
<?php

declare(strict_types=1);

namespace App\Internal\Domain\Entities;

use App\Internal\Domain\ValueObjects\PurchaseCategory;
use App\Internal\Domain\ValueObjects\Region;
use DateTimeImmutable;

final readonly class Purchase
{
    public function __construct(
        public string $purchaseId,
        public Region $region,
        public PurchaseCategory $category,
        public DateTimeImmutable $purchaseTimestamp,
    ) {
    }

    public function toArray(): array
    {
        return [
            'purchase_id' => $this->purchaseId,
            'region' => $this->region->value,
            'category' => $this->category->value,
            'purchase_timestamp' => $this->purchaseTimestamp->format(DATE_ATOM),
        ];
    }
}
